Add steps to migrate from Steeltoe 3.x to 4.0 (#402)
SteeltoeOSS/Documentation: docs/docs/v4/welcome/whats-new.md

- Actuators can be turned on/off or bound to different verbs at runtime using configuration
- Simplified content negotiation; updated all actuators to support latest Spring media type
- New actuator `/beans` that lists the contents of the .NET dependency container, including support for keyed services
- Update health checks and actuator to align with latest Spring
  - Hide components/details by default
  - Liveness/readiness contributors are turned off by default
  - Contributors can be turned on/off at runtime using configuration

>   - Contributors must be singletons (you can inject `IHttpContextAccessor`, but `HttpContext` is unavailable from Discovery)

Author: Contributors were previously registered as scoped by default, allowing request-dependent services to be injected. They are singletons now, so to access something like the query string parameters that initiated the health check, it requires injecting `IHttpContextAccessor`. The lifetime changed because a health check doesn't always execute during an HTTP request, for example, when invoked by our Eureka client (so then `HttpContext` is unavailable).

Assuming that answers your question, can you propose a better text?

>   - Contributors must be singletons (you can inject `IHttpContextAccessor`, but `HttpContext` is unavailable from Discovery)

Reviewer: OK, so the Discovery health contributor is an example of potentially necessary changes, but users don't need to make changes unless they're accessing our contributors or providing their own?

>   - Contributors must be singletons (you can inject `IHttpContextAccessor`, but `HttpContext` is unavailable from Discovery)

Author: The change is that health contributors used to be registered as scoped, which changed to singleton. Because the aggregator doesn't instantiate a scope before invoking actuators, they must be singletons now.

- Actuators can be turned on/off or bound to different verbs at runtime using configuration
- Simplified content negotiation; updated all actuators to support latest Spring media type
- New actuator `/beans` that lists the contents of the .NET dependency container, including support for keyed services
- Update health checks and actuator to align with latest Spring
  - Hide components/details by default
  - Liveness/readiness contributors are turned off by default
  - Contributors can be turned on/off at runtime using configuration

>   - Contributors must be singletons (you can inject `IHttpContextAccessor`, but `HttpContext` is unavailable from Discovery)

Reviewer: ```suggestion
- Health contributors must be singletons (you can still inject `IHttpContextAccessor`, but `HttpContext` is not available anymore)
```
Outdent on the assumption this piece isn't part of Spring compatibility
